storage/zynq7000-flash/zynq7000-flash.c

```diff
@@ -43,7 +43,7 @@ static ssize_t flash_read(oid_t *oid, offs_t offs, void *buff, size_t len)
 	ssize_t res = -ENOSYS;
 	storage_t *strg = storage_get(GET_STORAGE_ID(oid->id));
 
-	if (strg == NULL || strg->dev == NULL || (offs + len) >= strg->size) {
+	if (strg == NULL || strg->dev == NULL || (offs + len) >= strg->size || buff == NULL) {
```

Comment:
In case of `len` zero `buff` shouldn't matter - but I don't know if this is a use case. I think you can achieve it by doing `write(NULL, fd, 0)`, valid call - we should get `0` from it, not `-1` and `EINVAL`